doc/adr/0014-marconi-query-interface.rst

```diff
@@ -0,0 +1,39 @@
+ADR 14: Marconi Query Interface
+======================================================
+
+Date: 2022-10-27
+
+Author(s)
+---------
+
+Kayvan Kazeminejad <kayvan.kazeminejad@iohk.io>
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+Some DApps need to index and query a portion of the blockchain for a specific list of addresses. `Marconi <https://plutus-apps--763.org.readthedocs.build/en/763/adr/0004-marconi-initiative.html>`_ provides a general solution for indexing the blockchain data.  The query interface adds reporting capabilities on top of Marconi for a specific list of addresses for both Haskell and non-Haskell applications.
```

Comment:
I wouldn't use the first sentence, as the indexers are not limited to addresses.

How about something like (feel free to edit it):

`Marconi <https://plutus-apps--763.org.readthedocs.build/en/763/adr/0004-marconi-initiative.html>`_ provides a general solution for indexing the blockchain data.
However, the proposed query interface, which can fetch data stored on disk as well as data stored in memory, can only be used by another Haskell application.
Thus, we want to have a query interface which can query this data for both Haskell and non-Haskell applications.